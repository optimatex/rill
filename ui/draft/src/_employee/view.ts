export {
  React,
  useState,
  useCallback,
  css,
  styled,
  createGlobalStyle,
} from '@rill/view';
