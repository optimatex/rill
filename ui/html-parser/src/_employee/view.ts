export { React, RC } from '@rill/view';
