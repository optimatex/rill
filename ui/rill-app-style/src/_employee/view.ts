export { createGlobalStyle, css } from '@rill/view';
