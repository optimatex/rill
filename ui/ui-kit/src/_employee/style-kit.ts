export {
  // appLayout
  appLayoutCss,
  screenCss,
  sidebarColumnCss,
  flatColumnCss,
  mainColumnCss,
  // layout
  listCss,
  // Brand
  logCss,
  dualLogoCss,
  // Controller
  trivialAppHeaderCss,
  // _menu
  menuWrapCss,
  menuItemCss,
  // Image
  imageCss,
  imagesListCss,
  imagesGridCss,
  imagesLayoutItemCss,
  imageCropperCss,
  // Tags
  tag,
  tagsList,
  // Ui-primal
  // _grid
  gridRowContainerCss,
  gridrillRowContainerCss,
  gridColumnCss,
  // _card
  cardContainerCss,
  postContainerCss,
  postTitleCss,
  panelCss,
  // _buttons
  buttonCss,
  buttonActiveCss,
  buticonCss,
  // _form
  formContainerCss,
  formFieldFlatCss,
  formFieldColumnCss,
  formFieldErrorCss,
  formLabelTextCss,
  inputCss,
  toggleContainerCss,
  toggleContainerActiveCss,
  toggleRoundCss,
  toggleRoundActiveCss,
  // _article
  paragraphCss,
  nameCss,
  richContentWrapCss,
  // _table
  tableContainerCss,
  tableHeadCss,
  tableRowCss,
  tableColumnCss,
  // Unknown
  // _object
  stringValueCss,
  numberValueCss,
  booleanValueCss,
  emptyValueCss,
  arrayWrapCss,
  arrayItemCss,
  objectKeyCss,
  objectWrapCss,
  objectPairCss,
  // Unsorted
  lightIndicatorCss,
  popoverCss,
  pipeCss,
  hintableLogCss,
  fixedPanelCss,
} from '@rill/style-kit';
