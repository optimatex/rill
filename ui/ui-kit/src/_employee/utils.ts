export {
  toPairs,
  find,
  has,
  path,
  prop,
  curry,
  assoc,
  lensProp,
  view,
  set,
} from '@rill/utils';
