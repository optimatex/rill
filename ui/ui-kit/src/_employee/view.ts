export {
  React,
  useState,
  useCallback,
  // styled-components
  styled,
  // Lens
  // getTestId,
  // Types
  FC,
  RC,
  ReactNode,
  LegacyRef,
  // PropTypes
  IContainerProps,
  // Event types
  FormEvent,
  MouseEvent,
} from '@rill/view';
