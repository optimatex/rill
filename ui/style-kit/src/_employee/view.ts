import { css as originalCss } from '@rill/view';

export const css = originalCss;
