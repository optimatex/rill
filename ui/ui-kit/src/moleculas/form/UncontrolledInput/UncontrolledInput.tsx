import { FC, styled } from '_employee/view';
import { inputCss } from '_employee/style-kit';

import { uncontrolledInputDefaultProps, IUncontrolledInputProps } from '@types';

export const UncontrolledInput = styled.input`
  ${inputCss}
` as FC<IUncontrolledInputProps>;

UncontrolledInput.defaultProps = uncontrolledInputDefaultProps;
