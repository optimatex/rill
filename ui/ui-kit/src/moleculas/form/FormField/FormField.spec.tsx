import { React } from '_employee/view';
import { render } from 'test';
import { UncontrolledInput } from '../UncontrolledInput';
import { FormField } from '.';

describe('FormField', () => {
  it('should render without errors', () => {
    const { getByTestId, getByText, getAllByRole } = render(
      <FormField label="warrior" name="some-name" error="some error">
        <UncontrolledInput ref={null} name="test name" />
      </FormField>,
    );
    // container
    const container = getByTestId('field-some-name');
    expect(container).toMatchSnapshot();
    // error
    const error = getByText('some error');
    expect(error).toMatchSnapshot();
    // input
    const input = getAllByRole('textbox');
    expect(input).toMatchSnapshot();
    // label
    const label = getByText('warrior');
    expect(label).toMatchSnapshot();
  });
});
