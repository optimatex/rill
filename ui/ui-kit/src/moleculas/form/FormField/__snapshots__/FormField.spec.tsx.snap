// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField should render without errors 1`] = `
.c3 {
  box-sizing: border-box;
  padding: 4px 8px;
  border-width: 1px;
  border-style: solid;
  width: 100%;
}

.c3:hower {
  opacity: 0.8;
}

.c3:active {
  opacity: 0.9;
}

.c3:focus {
  opacity: 1;
}

.c1 {
  font: inherit;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c0 > * {
  margin-bottom: px;
}

.c0 > *:last-child {
  margin-bottom: 0;
}

.c2 {
  border-width: 1px;
  border-style: solid;
  box-sizing: border-box;
  padding: 4px 8px;
  position: absolute;
  top: 95%;
  left: 0%;
  -webkit-transform: translateX(15px);
  -ms-transform: translateX(15px);
  transform: translateX(15px);
  white-space: nowrap;
  background: red;
  color: #ffffff;
  z-index: 2;
}

<label
  class="c0"
  data-testid="field-some-name"
>
  <span
    class="c1"
    for="some-name"
  >
    warrior
  </span>
  <span
    class="c2"
  >
    some error
  </span>
  <input
    class="c3"
    name="test name"
    placeholder=""
    type="text"
    value=""
  />
</label>
`;

exports[`FormField should render without errors 2`] = `
<span
  class="sc-iwyYcG bmbXkw"
>
  some error
</span>
`;

exports[`FormField should render without errors 3`] = `
Array [
  <input
    class="sc-bdfBwQ gEwBtF"
    name="test name"
    placeholder=""
    type="text"
    value=""
  />,
]
`;

exports[`FormField should render without errors 4`] = `
<span
  class="sc-fKFyDc fYCVZm"
  for="some-name"
>
  warrior
</span>
`;
